modules/internal/ChapelRange.chpl

```diff
@@ -215,12 +215,21 @@ module ChapelRange {
     compilerError("Bounds of '..' must be integers of compatible types, when specified.");
   }
 
-  // Range builders for partially bounded ranges
-  proc chpl_build_partially_bounded_range(param bt: BoundedRangeType, bound: int(?w))
+  // Range builders for low bounded ranges
+  proc chpl_build_low_bounded_range(param bt: BoundedRangeType, bound: int(?w))
     return new range(int(w), bt, false, bound, bound);
-  proc chpl_build_partially_bounded_range(param bt: BoundedRangeType, bound: uint(?w))
+  proc chpl_build_low_bounded_range(param bt: BoundedRangeType, bound: uint(?w))
     return new range(uint(w), bt, false, bound, bound);
-  proc chpl_build_partially_bounded_range(param bt: BoundedRangeType, bound) {
+  proc chpl_build_low_bounded_range(param bt: BoundedRangeType, bound) {
+    compilerError("Bounds of '..' must be integers of compatible types, when specified.");
+  }
+
+  // Range builders for high bounded ranges
+  proc chpl_build_high_bounded_range(param bt: BoundedRangeType, bound: int(?w))
+    return new range(int(w), bt, false, bound, bound);
```

Comment:
It seems clear to me that you haven't changed anything here from what we were doing before, but it surprises me that 'bound' is sent in twice in the range constructors for the partially bounded cases (which would seem to store the value as both low and high, is that right?).  Is that just the way of the world?  I can't seem to recall...